DEVELOPER_GUIDE.md

```diff
@@ -78,6 +83,44 @@ Additionally, the `gcc` toolchain needs to be installed on Mac. To install, run:
 brew install gcc
 ```
 
+#### Extra setup for Mac M1 Machines
+
+The following commands enable running/building k-NN on M1 machines:
+
+```bash
+// Go to jni folder
+cd k-NN/jni
+
+// File changes required
+sed -i -e 's/\/usr\/local\/opt\/libomp\//\/opt\/homebrew\/opt\/llvm\//g' CMakeLists.txt
+sed -i -e 's/-march=native/-mcpu=apple-m1/g' external/nmslib/similarity_search/CMakeLists.txt
+sed -i -e 's/pragma message WARN/pragma message /g' external/nmslib/similarity_search/src/distcomp_scalar.cc
+sed -i -e 's/-mcpu=apple-a14/-mcpu=apple-m1/g' external/nmslib/python_bindings/setup.py
+sed -i -e 's/__aarch64__/__undefine_aarch64__/g' external/faiss/faiss/utils/distances_simd.cpp
+
+// Install llvm
+brew install llvm
+echo 'export PATH="/opt/homebrew/opt/llvm/bin:$PATH"' >> ~/.zshrc
+source ~/.zshrc
+
+// Set compiler path for CMAKE
+export CC=/opt/homebrew/opt/llvm/bin/clang
+export CXX=/opt/homebrew/opt/llvm/bin/clang++
+
+// Build
+cmake . --fresh
+make
+```
+
+Next, obtain a minimum distribution tarball of the k-NN version you want to build:
+
+1. Fork the [OpenSearch Repo](https://github.com/opensearch-project/OpenSearch) into your github account.
+2. Clone the repository locally
+3. Run the following commands:
+```cd OpenSearch && ./gradlew -p distribution/archives/darwin-tar assemble```
+4. You should see a opensearch-min-<version>-SNAPSHOT-darwin-x64.tar.gz file present in distribution/archives/darwin-tar/build/distributions/
+5. Build k-NN by passing the OpenSearch distribution path in `./gradlew <integTest/run> -PcustomDistributionUrl="<Full path to .tar.gz file you noted above>"`
```

Comment:
do we need customDistribution here? does OpenSearch is not providing the distribution for M1Mac?

Reply:
Yeah it throws an error if you don't specify the exact path to the distribution:
```
Execution failed for task ':run'.
> Could not resolve all files for configuration ':opensearch_distro_extracted_testclusters--integTest-0-3.0.0-SNAPSHOT-'.
   > Could not resolve opensearch-distribution-snapshot:opensearch:3.0.0-SNAPSHOT.
     Required by:
         project :
      > Could not resolve opensearch-distribution-snapshot:opensearch:3.0.0-SNAPSHOT.
         > Could not get resource 'https://artifacts.opensearch.org/snapshots/core/opensearch/3.0.0-SNAPSHOT/opensearch-min-3.0.0-SNAPSHOT-darwin-arm64-latest.tar.gz'.
            > Could not HEAD 'https://artifacts.opensearch.org/snapshots/core/opensearch/3.0.0-SNAPSHOT/opensearch-min-3.0.0-SNAPSHOT-darwin-arm64-latest.tar.gz'. Received status code 403 from server: Forbidden

```

Reply:
Thanks make sense. You can resolve this comment, but before that can you just create a github issue on the OpenSearch core repo to start creating the min distribution for M1Mac.

Reply:
Could you try this? https://github.com/opensearch-project/k-NN/issues/580#issuecomment-1741272382

Reply:
That link is just for the tarball, so we could provide the download.  However, the user would still have to specify the path to that download

Reply:
https://github.com/opensearch-project/OpenSearch/issues/10593